pyqtgraph/graphicsItems/LegendItem.py

```diff
@@ -73,16 +75,39 @@ def __init__(self, size=None, offset=None, horSpacing=25, verSpacing=0,
         if size is not None:
             self.setGeometry(QtCore.QRectF(0, 0, self.size[0], self.size[1]))
 
+        if sampleType is not None:
+            assert issubclass(sampleType, GraphicsWidget)
+            self.sampleType = sampleType
+        else:
+            self.sampleType = ItemSample
+
         self.opts = {
             'pen': fn.mkPen(pen),
             'brush': fn.mkBrush(brush),
             'labelTextColor': labelTextColor,
             'labelTextSize': labelTextSize,
             'offset': offset,
         }
-
         self.opts.update(kwargs)
 
+    def setSampleType(self, sample):
+        """Set the new sample item claspes"""
+        if sample == self.sampleType:
```

Comment:
do we want an equality operator here or an `is` operator?

Reply:
i think its a good idea